### PR TITLE
feat(ShowMore): extend ShowMore to display line breaks

### PR DESCRIPTION
**Pull request by vshyrokov**
### Description

Added ability to show line breaks

Go to https://picasso.toptal.net/add-line-breaks-into-show-more/?path=/story/components-showmore--showmore#line-breaks and check that text with line breaks correctly rendered in the Show More component

### Screenshots

| Before.                                 | After.                                  |
| --------------------------------------- | --------------------------------------- |
|image|image|

### Review

- [x] Read [CONTRIBUTING.md](https://github.com/toptal/picasso/blob/master/CONTRIBUTING.md) and [Component API principles](https://github.com/toptal/picasso/blob/master/docs/contribution/component-api.md)
- [x] Annotate all `props` in component with documentation
- [x] Create `examples` for component
- [x] Ensure that deployed demo has expected results and good examples
- [x] Ensure that tests pass by running `yarn test`
- [x] Ensure that visuals tests pass by running `yarn test:visual`. If not - check the documentation [how to fix visual tests](https://github.com/toptal/picasso/blob/master/docs/contribution/visual-testing.md#fixing-broken-visual-tests-inside-a-pr)
- [x] Ensure the changed/created components have not caused accessibility issues. [How to use accessibility plugin in storybook](https://github.com/toptal/picasso/blob/master/docs/contribution/accessibility.md).

<details>
<summary>PR commands</summary>
<br />

List of available commands:

- `@toptal-bot run all` - Run whole pipeline
- `@toptal-bot run build` - Check build
- `@toptal-bot run visual` - Run visual tests
- `@toptal-bot run deploy:documentation` - Deploy documentation
- `@toptal-bot run package:alpha-release` - Release alpha version

</details>

<details>
<summary>PR Review Guidelines</summary>
<br />

#### When to approve? ✅

**You are OK** with merging this PR and

1. You have no extra requests.
2. You have optional requests.
   1. Add `nit:` to your comment. (ex. `nit: I'd rename this variable from makeCircle to getCircle`)

#### When to request changes? ❌

**You are not OK** with merging this PR because

1. Something is broken after the changes.
2. Acceptance criteria is not reached.
3. Code is dirty.

#### When to comment (neither ✅ nor ❌)

**You want your comments to be addressed** before merging this PR in cases like:

1. There are leftovers like unnecessary logs, comments, etc.
2. You have an opinionated comment regarding the code that requires a discussion.
3. You have questions.

#### How to handle the comments?

1. An owner of a comment is the only one who can resolve it.
2. An owner of a comment must resolve it when it's addressed.
3. A PR owner must reply with ✅ when a comment is addressed.

</details>

### CODE REVIEW

File: packages/picasso/src/ShowMore/utils.tsx [hunk 1]
```diff
@@ -0,0 +1,14 @@
+import React from 'react'
+
+export const replaceLineBreaksWithTags = (children: string) =>
```
Comment (OndrejTuma) [inline packages/picasso/src/ShowMore/utils.tsx:3]: I wouldn't count on children to be a string, we want to adopt react children as well in the future

File: packages/picasso/src/ShowMore/utils.tsx [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+  children.split('\n').map((line, index, { length }) => {
+    // eslint-disable-next-line react/no-array-index-key
+    const newLine = <span>{line}</span>
```
Comment (OndrejTuma) [inline packages/picasso/src/ShowMore/utils.tsx:6]: can you return only line to preserve content structure?

File: packages/picasso/src/ShowMore/utils.tsx (hunk 1 above)
Comment (vshyrokov) [reply packages/picasso/src/ShowMore/utils.tsx:3]: I think in this case we should add a check in the future regarding string and ReactNode, at the moment, in props, we have a string only

**Comment by vshyrokov** [comment]
@toptal-bot run all

File: packages/picasso/src/ShowMore/utils.tsx (hunk 1 above)
Comment (OndrejTuma) [reply packages/picasso/src/ShowMore/utils.tsx:3]: okay, make sense to adjust it when necessary 👍

File: packages/picasso/src/ShowMore/test.tsx [hunk 3]
```diff
@@ -135,4 +135,16 @@ describe('ShowMore', () => {
...
+
+  describe('when text with line breaks specified', () => {
+    it('should render span and br to respec line breaks', () => {
+      const { container } = renderShowMore({
+        children:
+          '1. More about contact\r\n\r\nlocation:\r\n\r\nblah blah blah blah blah\r\n\r\nblah blah blah blah blah\r\n\r\nblah blah blah blah blah\r\n\r\nblah blah blah\r\n\r\n2. More about contact\r\n\r\nlocation:\r\n\r\nblah blah blah blah blah\r\n\r\nblah blah blah blah blah\r\n\r\nblah blah blah blah blah\r\n\r\nblah blah blah\r\n\r\n1. More about contact\r\n\r\nlocation:\r\n\r\nblah blah blah blah blah\r\n\r\nblah blah blah blah blah\r\n\r\nblah blah blah blah blah\r\n\r\nblah blah blah',
+        initialExpanded: true
+      })
+
+      expect(container).toMatchSnapshot()
```
Comment (OndrejTuma) [inline packages/picasso/src/ShowMore/test.tsx:147]: I'd actually create a cypress test with happoScreenshot rather than snapshot here.  wdyt?

**Comment by OndrejTuma** [comment]
Changeset is missing ☝️

File: cypress/component/ShowMore.spec.tsx [hunk 4]
```diff
@@ -0,0 +1,58 @@
+import React from 'react'
+import { ShowMore } from '@toptal/picasso'
+import { mount } from '@cypress/react'
+import { TestingPicasso } from '@toptal/picasso/test-utils'
+
+describe('ShowMore', () => {
+  it('renders expanded ShowMore with line breaks', () => {
+    const text =
```
Comment (OndrejTuma) [inline cypress/component/ShowMore.spec.tsx:8]: text constant can be hoisted to a global context

File: .changeset/curly-boxes-hug.md [hunk 5]
```diff
@@ -0,0 +1,5 @@
+---
+'@toptal/picasso': minor
+---
+
+Extended Show More to display line breaks
```
Comment (OndrejTuma) [inline .changeset/curly-boxes-hug.md:5]: I'd prefer something like:
```suggestion
`ShowMore` component now displays line breaks in truncated state
```
We stopped writing progress report pages in favor of more explicit changelogs 🙇

**Review by OndrejTuma — COMMENTED** [review]
Two tiny things and we're there! 🎉

**Review by OndrejTuma — APPROVED** [review]

**Review by inemiro — APPROVED** [review]
Looks good from my POV 👍

File: packages/picasso/src/ShowMore/utils.tsx [hunk 6]
```diff
@@ -0,0 +1,16 @@
+import React from 'react'
+
+export const replaceLineBreaksWithTags = (children: string) =>
+  children.split('\n').length > 1
+    ? children.split('\n').map((line, index, { length }) => {
```
Comment (dmaklygin1) [inline packages/picasso/src/ShowMore/utils.tsx:5]: please, cache the `children.split('\n')` in a variable

File: packages/picasso/src/ShowMore/ShowMore.tsx [hunk 7]
```diff
@@ -57,7 +58,9 @@ export const ShowMore = forwardRef<HTMLSpanElement, Props>(function ShowMore(
         className={className}
         style={style}
       >
-        <Truncate lines={!shownMore && rows}>{children}</Truncate>
+        <Truncate lines={!shownMore && rows}>
+          {replaceLineBreaksWithTags(children)}
```
Comment (dmaklygin1) [inline packages/picasso/src/ShowMore/ShowMore.tsx:62]: I guess, this render isn't memoized

File: packages/picasso/src/ShowMore/utils.tsx [hunk 8]
```diff
@@ -0,0 +1,16 @@
+import React from 'react'
+
+export const replaceLineBreaksWithTags = (children: string) =>
```
Comment (dmaklygin1) [inline packages/picasso/src/ShowMore/utils.tsx:3]: I would specify a return type of this function

**Review by dmaklygin1 — APPROVED** [review]

File: .changeset/curly-boxes-hug.md [hunk 9]
```diff
@@ -0,0 +1,5 @@
+---
+'@toptal/picasso': minor
+---
+
+`ShowMore` component now displays line breaks in truncated state
```
Comment (denieler) [inline .changeset/curly-boxes-hug.md:5]: @vshyrokov not a correct changeset format according to https://github.com/toptal/picasso/blob/master/docs/contribution/changeset-guidelines.md

**Pull request merged**